src/Microsoft.OData.Core/Uri/NodeToStringBuilder.cs

```diff
@@ -470,7 +471,240 @@ internal String TranslateNode(QueryNode node)
+        /// <returns>The translated string with $it binding.</returns>
+        private string BindNode(QueryNode node, ResourceRangeVariable filterClauseRangeVariable)
+        {
+            BinaryOperatorNode binaryNode = node as BinaryOperatorNode;
+            InNode inNode = node as InNode;
+            AnyNode anyNode = node as AnyNode;
+            SingleValueFunctionCallNode singleValueFunctionCallNode = node as SingleValueFunctionCallNode;
+            SingleValueOpenPropertyAccessNode singleValueOpenPropertyAccessNode = node as SingleValueOpenPropertyAccessNode;
+
+            if (binaryNode != null)
+            {
+                return BindBinaryOperatorNode(binaryNode, filterClauseRangeVariable);
+            }
+            else if (inNode != null)
+            {
+                return BindInNode(inNode, filterClauseRangeVariable);
+            }
+            else if (anyNode != null)
+            {
+                return BindAnyNode(anyNode, filterClauseRangeVariable);
+            }
+            else if (singleValueFunctionCallNode != null)
+            {
+                return BindSingleValueFunctionCallNode(singleValueFunctionCallNode, filterClauseRangeVariable);
+            }
+            else if (singleValueOpenPropertyAccessNode != null)
+            {
+                return BindSingleValueOpenPropertyAccess(singleValueOpenPropertyAccessNode, filterClauseRangeVariable);
+            }
+            else
+            {
+                return this.TranslateNode(node);
+            }
+        }
+
+        /// <summary>
+        /// Bind $it to the <see cref="BinaryOperatorNode"/> translated string.
+        /// </summary>
+        /// <param name="node">node to bind.</param>
+        /// <param name="filterClauseRangeVariable">The <see cref="FilterClause"/> range variable.</param>
+        /// <returns>The translated string with $it binding.</returns>
+        private string BindBinaryOperatorNode(BinaryOperatorNode node, ResourceRangeVariable filterClauseRangeVariable)
+        {
+            string translatedNode = this.TranslateNode(node);
+            ResourceRangeVariableReferenceNode leftRangeVariableNode = GetResourceRangeVariableReferenceNode(node.Left);
+            ResourceRangeVariableReferenceNode rightRangeVariableNode = GetResourceRangeVariableReferenceNode(node.Right);
+
+            BinaryOperatorKind binaryNodeKind = node.OperatorKind;
+
+            string binaryOperator = this.BinaryOperatorNodeToString(binaryNodeKind);
+            string[] binarySeparator = { binaryOperator };
+            string[] substrings = translatedNode.Trim().Split(binarySeparator, StringSplitOptions.RemoveEmptyEntries);
+            string leftBinary = substrings[0].Trim();
+            string rightBinary = substrings[1].Trim();
+
+            if (leftRangeVariableNode != null && IsDifferentSource(filterClauseRangeVariable, leftRangeVariableNode))
+            {
+                leftBinary = "$it/" + leftBinary;

```

Comment:
Maybe have `ItSlash` as a constant in `ExpressionConstants` class?

Reply:
Replaced it with `ExpressionConstants.It + ExpressionConstants.SymbolForwardSlash`

Reply:
For perf sensitive scenarios it would be better to define an ItSlash = ExpressionConstants.It + ExpressionConstants.SymbolForwardSlash, but a single call per range variable translation is not going to be a hotspot in constructing the Uri so I wouldn't worry about it here.

---
In reply to: [574337676](https://github.com/OData/odata.net/pull/1991#discussion_r574337676) [](ancestors = 574337676)